src/jit/lclvars.cpp

```diff
@@ -235,6 +235,52 @@ void Compiler::lvaInitTypeRef()
 
     lvaInitArgs(&varDscInfo);
 
+#if FEATURE_FASTTAILCALL
+
+    //-------------------------------------------------------------------------
+    // Calculate the argument register usage.
+    //
+    // This will later be used for fastTailCall determination
+    //-------------------------------------------------------------------------
+
+    unsigned argRegCount      = 0;
+    unsigned floatingRegCount = 0;
+    unsigned stackArgCount    = 0;
+    size_t   stackSize        = 0;
+    unsigned compArgCount     = info.compArgsCount;
+
+    auto incrementRegCount = [&floatingRegCount, &argRegCount](LclVarDsc* varDsc) {
+        varDsc->IsFloatRegType() ? ++floatingRegCount : ++argRegCount;
+    };
```

Comment:
Also for the HFA case I had overlooked it. Upon investigation on why it did not cause problems I found that all HFAs are marked as address exposed on Arm64 therefore, this code path will never return curDsc->lvIsHfa().

I added an assert to catch the case when we remove the HFA address exposed later.